service/api_logic/interactions_logic.py

```diff
@@ -0,0 +1,75 @@
+from dto.api_output import OutputInteractions
+from exept.exeptions import IncorrectInteractionType, BlobFetchError, DatabaseConnectionError
+from service.api_logic.models.api_models import InteractionTypes
+from database.postgres.dto import InteractionWithNewsDTO
+from logger.logger import Logger
+from datetime import datetime
+
+LIKE_ID = InteractionTypes['LIKE'].value
+DISLIKE_ID = InteractionTypes['DISLIKE'].value
+
+class InteractionWithNewsService:
+    def __init__(self, interaction_with_news_dal, news_dal):
+        self._interaction_with_news_dal = interaction_with_news_dal
+        self._logger = Logger("logger", "all.log").logger
+        self._news_dal = news_dal
+
+
+    def __convert_input_to_db_dto(self, interaction_input_dto):
+        try:
+            interaction_type_id = InteractionTypes[interaction_input_dto.interaction_type.upper()].value
+            news_entry = self._news_dal.get_news_by_id(interaction_input_dto.blob_id)
+
+            if not news_entry:
+                raise BlobFetchError(interaction_input_dto.blob_id)
+
+            db_dto = InteractionWithNewsDTO(
+                news_id=news_entry.news_id,
+                user_id=interaction_input_dto.user_id,
+                interaction_type=interaction_type_id,
+                timestamp=datetime.now()
+            )
+
+            return db_dto
+
+        except KeyError:
```

Comment:
If that KeyError is focused on `blob_id` alone, it'd be better to just check that `blob_id` explicitly (some `if` statement or whatever) than catching an exception. Cleaner approach and better performance as well.

Basically:

It's better to do `if (true)` -> raise Exception
than `try ... except ... raise`